### PR TITLE
RoR group project 

### PR DESCRIPTION
**Pull request by abdullah-asghar-khan**
**Description**

The Recipe app keeps track of all your recipes, ingredients, and inventory. It will allow you to save ingredients, keep track of what you have, create recipes, and generate a shopping list based on what you have and what you are missing from a recipe.

**Interfaces**

- A login page.

- A registration page.

- A food list (with all CRUD implementation, except for 'update').

- A recipe list (with all CRUD implementation, except for 'update').

- A general shopping list view (all missing food for all your recipes and total price).

- A list of all public recipes from other users with their names and total prices.

**Project requirements**

- Followed the layout of the wireframes provided. Personalized the rest of the design including colors, typographies, spacings, etc.

- Login page and registration page:

1. Built with Devise.

- Food list:

1. Displays a list of food added by the logged-in user as in the wireframe and displays also quantity of a given food.

2. Leads to a form that allows users to add new food.

- Recipes list:

1. Displays a list of recipes created by the logged-in user as in the wireframe.

2. Leads to recipe details.

3. If the user is the owner of the recipe, removing the recipe is allowed.

- Public recipe list:

1. Displays a list of all public recipes ordered by the newest as in the wireframe.

2. Leads to recipe details.

- Recipe details:

1. If the recipe is public or the user is the owner of the recipe, the page displays the recipe details as in the wireframe.

2. If the user is the owner of the recipe, there is a button that leads to the form that allows the user to add new food.

- General shopping list view:

1. Displays the list of food that is missing for all recipes of the logged-in user (comparing the list of food for all recipes with the general food list of that user).

2. Counts the total food items and total price of the missing food.

- Made sure there are no N+1 queries happening.

- Created a navigation menu that allows users to open all of the pages inside the app.

- Technical setup

1. Set up the repository on GitHub and use Gitflow.

2. Set up Devise for authentication.

P.S. The technical test setup is corrected and fixed, and some tests are added to the project, but it is not a requirement.

### CODE REVIEW

**Review by youmari — COMMENTED** [review]
Hi @Abdullah2213565 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that should be opened form `dev` to `main` in order to achieve the git-flow 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/models/food_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  it 'measurment_unit field should not be blank' do
+    subject.measurment_unit = 'kilograms'
+    expect(subject.measurment_unit).to eq 'kilograms'
+  end
+
+  it 'price must be an integer greater than or equal to 0' do
+    subject.price = 3
+    expect(subject.price).to be >= 0
+  end
+end
```
Comment (youmari) [inline spec/models/food_spec.rb:22]: - [ ] your tests are failing please make sure to fix that in order for them to pass 
Screen Shot 2022-11-10 at 19 16 25

**Review by youmari — CHANGES_REQUESTED** [review]
Hi @Abdullah2213565 @AsimKhan2019 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
✔️ PRO PR
✔️ No linter error

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/food_spec.rb (hunk 1 above)
Comment (AsimKhan2019) [reply spec/models/food_spec.rb]: @youmari I just updated the Gem File and made a fresh install of the RSpec, the food model RSpec test file is now running.

**Review by danmainah — APPROVED** [review]
Hi @team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AsimKhan2019** [comment]
@danmainah Thank you for the approval

**Pull request merged**